lib/dateTime.js

```diff
@@ -1167,3 +1167,38 @@ export function formatTimeFromTimestamp(timestamp, options) {
 	const date = parseLocalDateTimeFromTimestamp(timestamp);
 	return formatTime(date, options);
 }
+
+export function formatRelativeDateTime(date, { numeric = 'always', style = 'long' } = {}) {
+
+	const now = new Date();
+	const then = new Date(date);
+
+	if (!Intl.RelativeTimeFormat) {
+		return now.toDateString() === then.toDateString() ? formatTime(then) : formatDate(then);
+	}
+
+	const rtf = new Intl.RelativeTimeFormat(getLanguage(), {
+		localeMatcher: 'best fit',
+		numeric,
+		style
+	});
+
+	const delta = (then.getTime() - now.getTime()) / 1000;
+
+	const unitCutoffs = {
+		second: 60,
+		minute: 60,
+		hour: 24,
+		day: 7,
+		week: 4.348,
+		month: 12
+	};
+
+	const { value, unit = 'year' } = Object.entries(unitCutoffs).reduce((acc, [unit, cutoff], idx, arr) => {
+		if (Math.round(Math.abs(acc.value)) >= Math.round(cutoff)) return { value: acc.value / cutoff };
+		arr.splice(idx + 1); // short-circuit
+		return { value: acc.value, unit };
+	}, { value: delta });
+
+	return rtf.format(Math.round(value), unit);
+}
```

Comment:
Just going to put this here so we can discuss in a thread. Here's a gap analysis between what's here and the "fuzzy date" stuff in the LMS.

Links:
- [server-side rendering logic for plain text](https://search.d2l.dev/xref/lms/lp/framework/core/D2L.LP/LP/TextProcessing/Rendering/Renderers/FuzzyLocalDateTimePlainTextRenderer.cs?r=b514705c#45)
- [server-side rendering logic for HTML using abbr tag](https://search.d2l.dev/xref/lms/lp/framework/core/D2L.LP/LP/TextProcessing/Rendering/Renderers/FuzzyLocalDateTimeHtmlRenderer.cs?r=b514705c#13)
- [client-side "live" updating](https://search.d2l.dev/xref/lms/lp/framework/web/D2L.LP.Web/UI/JavaScript/Globalization/Formatting/DateTime/FuzzyDate.js?r=a540d29f)

The main thing that isn't really a gap but more something we'd need to build on top of this in `@brightspace-ui/core` would be a wrapper web component that would add:
1.  "live" updating while the user stays on the page. So it might start out saying "just now" but after a minute would change to "1 minute ago", and so on. For that to be super efficient, it might be helpful for there to be an advanced API here that returns how long it should wait before checking back in for a new value.
2. An `<abbr>` element (or similarly accessible alternative) that provides to full date & time in addition to the relative value.

The logic in the LMS code is slightly different than what's here:

| Threshold | Old Behaviour | New Behaviour |
| ------------- | ------------- | ------------- |
| future values | not supported | New implementation supports future values, e.g. "in 1 minute" |
| < 30 seconds  | "just now"  | This implementation doesn't have the concept of "just now"... it would say something like "3 seconds ago". Also, "just now" is more of a "in the past" thing and for a future thing that's arrived should likely be "right now", so maybe it's fine that we ignore it. |
| < 90 seconds  | "one minute ago"  | the difference here is the "one" vs. "1" -- seems fine |
| < 150 seconds | "two minutes ago" | again, compared to "2 minutes ago" -- also seems fine |
| < 45 minutes | "12 minutes ago" | matches the new stuff |
| < 90 minutes | "one hour ago" | will now say "x minutes ago" up until 60, then will say "1 hour ago" up until 90 |
| < 150 minutes | "two hours ago" | same logic as above |
| < 6 hours or same day | "13 hours ago" | new logic doesn't care about whether it's the same day or not, it'll always say "x hours ago" for 24 hours |
| yesterday | "yesterday at {time}" | will now be "x hours ago" up until 24, then "1 day ago" with no time. This is a somewhat significant change. |
| within 3 days | "{dayOfWeek} at {time}" | again, the old behaviour is friendlier and will say something like "Tuesday at 3:12 PM" whereas now it'll just be "2 days ago" |
| same year | "{monthAndDay} at "{time}" | similarly, much more precise compared to "3 weeks ago" or "5 months ago". Probably fine since this should be used for non-precise things? |
| not same year | normal date-time | new behaviour maxes out at months so it'll always just be something "39 months ago". Maybe we want to add a "year" bucket? |

IMO, I think it's worth it to ditch that old code in favour of this implementation, but wanted to spell it all out in case others disagree. There may also be a couple small enhancements we could make to this code to close the gap a bit without compromising the simplicity.

Reply:
Note it does do "years", but since it doesn't do anything more like "decades", "years" is not included in the `unitCutoffs` (i.e. it has no cutoff) and is assumed when the `reduce` returns no `unit`.

Reply:
We definitely could play with things more and use `numeric: auto` to get things like "yesterday". More complicated comparisons and conditions could get us "yesterday at {time}", but I think I prefer the less precise values (numeric or not), and a date and time on hover.

Thanks for breaking this down.

Reply:
Ah, understood -- I've updated the table to include the "years" thing.

Reply:
I'm good with these changes too, though I hope we can use the "yesterday" logic for >= 6 hours, as Danny confirmed. 

The other cases where we lose specificity are okay with me, I think it's more in keeping with the purpose of the relative date format. So "5 months ago" is less precise, but that's the point -- it's meant to answer the question "how long ago was this" rather than "when was this?", and the full date & time can and should still be available on hover/tap.